pkg/datastore/lrdatastore.go

```diff
@@ -0,0 +1,159 @@
+package datastore
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/flashbots/go-boost-utils/types"
+	"github.com/google/uuid"
+	"github.com/lthibault/log"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/blocknative/dreamboat/pkg/structs"
+)
+
+type LocalDatastore interface {
```

Comment:
We cannot go with the second option, because we need to differentiate local and remote explicitly—this way we know that we need to query local before remote, etc.